docs/user/modelling.md

```diff
@@ -0,0 +1,128 @@
+
+```python
+from nautobot.ipam.models import Prefix
+from nautobot_ssot.contrib import NautobotModel
+class PrefixModel(NautobotModel):
+    _model = Prefix
+    _identifiers = ("network", "prefix_length")
+    _attributes = ("location__name", "location__location_type__name")
+
+    network: str
+    prefix_length: int
+
+    location__name: str
+    location__location_type__name: str
+```
+
+Now, on model `create` or `update`, the SSoT framework will dynamically pull in the location with the specified name and location type name, uniquely identifying the location and populating the foreign key.
+
+### Special Case: Generic Foreign Key
+
+In the case of a [generic foreign key](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/), we are faced with a problem. Normally, the [content type](https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/) of a relationship field can be inferred from the model class. In the case of generic foreign keys however, this is not the case. In the example of the [IP address](https://docs.nautobot.com/projects/core/en/stable/models/ipam/ipaddress/?h=ip+address) model in Nautobot, the `assigned_object` field can point to either a device's or a VM's interface. We address this the following way:
+
+```python
+from nautobot.ipam.models import IPAddress
+from nautobot_ssot.contrib import NautobotModel
+class NautobotIPAddress(NautobotModel):
+    _model = IPAddress
+    _modelname = "ip_address"
+    _identifiers = (
+        "host",
+        "prefix_length",
+    )
+    _attributes = (
+        "status__name",
+        "assigned_object__app_label",
+        "assigned_object__model",
+        "assigned_object__device__name",
+        "assigned_object__name",
+    )
+
+    host: str
+    prefix_length: int
+    status__name: str
+    assigned_object__app_label: str
+    assigned_object__model: str
+    assigned_object__device__name: str
+    assigned_object__name: str
+```
+
+!!! warning
+    A limitation of this approach is that we are locked into a single kind of content type unless the field names for all content types match. In this specific case, the VM interface model does not have a `device` field. This is a known issue.
```

Comment:
we are locked into a single kind of content type unless the field names -> we are locked into a single kind of content type [per something] unless the field names